docs/running-on-kubernetes.md

```diff
@@ -990,6 +990,52 @@ See the [configuration page](configuration.html) for information on Spark config
   Specify whether executor pods should be deleted in case of failure or normal termination.
   </td>
 </tr>
+<tr>
+  <td><code>spark.kubernetes.kubernetesClient.submission.connectionTimeout</code></td>
+  <td>10000</td>
+  <td>
+    Connection timeout in milliseconds for the kubernetes client to use for starting the driver. In client mode, use
+    <code>spark.kubernetes.kubernetesClient.connectionTimeout</code> instead.
+  </td>
+</tr>
+<tr>
+  <td><code>spark.kubernetes.kubernetesClient.submission.requestTimeout</code></td>
+  <td>10000</td>
+  <td>
+    Request timeout in milliseconds for the kubernetes client to use for starting the driver. In client mode, use
+    <code>spark.kubernetes.kubernetesClient.submission.requestTimeout</code> instead.
+  </td>
+</tr>
+<tr>
+  <td><code>spark.kubernetes.kubernetesClient.driver.connectionTimeout</code></td>
+  <td>10000</td>
+  <td>
+    Connection timeout in milliseconds for the kubernetes client in driver to use when requesting executors.
+    In client mode, use <code>spark.kubernetes.kubernetesClient.connectionTimeout</code> instead.
```

Comment:
Can we have a more intuitive name instead of having explanation like this?

Currently, the names look a little confusing. Can we have some naming rule?
- `spark.kubernetes.kubernetesClient.sumbission.*` -> starting the driver
- `spark.kubernetes.kubernetesClient.driver.*` -> requesting executors (non-client mode)
- `spark.kubernetes.kubernetesClient.*` -> requesting executors (client mode)